File: src/main/java/com/vaadin/demo/component/icons/IconsAccessibility.java
package com.vaadin.demo.component.icons;

import com.vaadin.demo.flow.routing.Route;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("icons-accessibility")
public class IconsAccessibility extends Div {

  public IconsAccessibility() {
    // tag::snippet[]
    Button closeButton = new Button(VaadinIcon.CLOSE.create());
    closeButton.setAriaLabel("Close dialog");
    // end::snippet[]

    add(closeButton);
  }

  public static class Exporter extends DemoExporter<IconsAccessibility> { // hidden-source-line
  }
}

File: frontend/demo/component/icons/icons-accessibility.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import '@vaadin/button';
import '@vaadin/icon';
import '@vaadin/icons';

@customElement('icons-accessibility')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-button aria-label="Close dialog" theme="icon">
        <vaadin-icon icon="vaadin:close" slot="prefix"></vaadin-icon>
      </vaadin-button>
      <!-- end::snippet[] -->
    `;
  }
}
